src/session.rs

```diff
@@ -12,34 +12,82 @@ pub enum Session {
     Hmac(HmacSession),
     Policy(PolicySession),
     Trial(TrialSession),
+    Password,
 }
 
 impl Session {
-    pub fn handle(&self) -> SessionHandle {
-        match self {
-            Session::Hmac(session_object) => session_object.handle(),
-            Session::Policy(session_object) => session_object.handle(),
-            Session::Trial(session_object) => session_object.handle(),
+    ///
+    /// Function that creates a Option<Session>
+    ///
+    pub fn create(
+        session_type: SessionType,
+        session_handle: SessionHandle,
+        auth_hash: HashingAlgorithm,
+    ) -> Option<Session> {
```

Comment:
Is there any doc explaining why we use `Option`? It feels like we should mention that we map `ESYS_TR_NONE` to `Option::None` throughout the crate

Reply:
Yes, I will add a detail section that explains that.